src/main/java/gregtech/core/hooks/CTMModHooks.java

```diff
@@ -0,0 +1,36 @@
+package gregtech.core.hooks;
+
+
+import gregtech.api.util.GTLog;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.util.BlockRenderLayer;
+import team.chisel.ctm.api.model.IModelCTM;
+import team.chisel.ctm.client.model.ModelCTM;
+
+import java.lang.reflect.Field;
+
+public class CTMModHooks {
+    private static Field layers;
+
+    static{
+        try {
+            layers = ModelCTM.class.getDeclaredField("layers");
+            layers.setAccessible(true);
+        } catch (NoSuchFieldException e) {
+            GTLog.logger.error("CTMModHooks no such field");
+        }
+    }
+
+    public static boolean canRenderInLayer(IModelCTM model, IBlockState state, BlockRenderLayer layer) {
+        boolean flag = model.canRenderInLayer(state, layer);
+        if (model instanceof ModelCTM && layers != null) {
+            try {
+                return CTMHooks.checkLayerWithOptiFine(flag, layers.getByte(model), layer);
+            } catch (Exception ignored) {
+                layers = null;
+                GTLog.logger.error("CTMModHooks Field error");
```

Comment:
Given how often this would be hit on servers, this would immediately spam the logs. For that matter, this whole function here could probably be optimized, and that model.canRenderInLayer seems a bit unsafe to not run in the try-catch.

Reply:
layers is set to `null` . Messages are sent only once

Reply:
That sounds like that would _immediately_ cause an NPE on the assignment of `flag` up above.

Reply:
Wait no that's layer, disregard that. But still, that first call seems really strange to put there.

Reply:
why..... did you get the crash report?

Reply:
yeah, layer and layers. in addition, Fields must be accessed by a try/catch.  This is not hit frequent and only hit when compiled the chunk.

Reply:
So model.canRenderInLayer works fine on a server?

Reply:
After testing this, it looks like this bit of code works without a crash.